File: algorithms/2_medium/L/03719_longest_balanced_subarray_i/bruteforce.cpp
class Solution {
public:
    int longestBalanced(vector<int>& nums) {
        const size_t len{ nums.size() };
        size_t longest{ 0 };

        for (size_t i{ 0 }; i < len; ++i) {
            unordered_set<int> evens;
            unordered_set<int> odds;

            for (size_t j{ i }; j < len; ++j) {
                if ((nums[j] & 1) == 0) {
                    odds.insert(nums[j]);
                } else {
                    evens.insert(nums[j]);
                }

                if (odds.size() == evens.size()) {
                    longest = std::max(longest, j - i + 1);
                }
            }
        }

        return static_cast<int>(longest);
    }
};

File: algorithms/2_medium/L/03719_longest_balanced_subarray_i/bruteforce.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const longestBalanced = (nums) => {
    const len = nums.length;
    let longest = 0;

    for (let i = 0; i < len; ++i) {
        const evens = new Set();
        const odds = new Set();

        for (let j = i; j < len; ++j) {
            if ((nums[j] & 1) === 0) {
                odds.add(nums[j]);
            } else {
                evens.add(nums[j]);
            }

            if (odds.size === evens.size) {
                longest = Math.max(longest, j - i + 1);
            }
        }
    }

    return longest;
};
